static/js/popover_menus.js

```diff
@@ -87,4 +107,196 @@ export function initialize() {
+                editability_menu_item = $t({defaultMessage: "View source"});
+            }
+            const topic = message.topic;
+            const can_mute_topic =
+                message.stream && topic && !muted_topics.is_topic_muted(message.stream_id, topic);
+            const can_unmute_topic =
+                message.stream && topic && muted_topics.is_topic_muted(message.stream_id, topic);
+
+            const should_display_edit_history_option =
+                message.edit_history &&
+                message.edit_history.some(
+                    (entry) =>
+                        entry.prev_content !== undefined ||
+                        util.get_edit_event_prev_topic(entry) !== undefined,
+                ) &&
+                page_params.realm_allow_edit_history;
+
+            // Disabling this for /me messages is a temporary workaround
+            // for the fact that we don't have a styling for how that
+            // should look.  See also condense.js.
+            const should_display_collapse =
+                !message.locally_echoed && !message.is_me_message && !message.collapsed;
+            const should_display_uncollapse =
+                !message.locally_echoed && !message.is_me_message && message.collapsed;
+
+            const should_display_edit_and_view_source =
+                message.content !== "<p>(deleted)</p>" ||
+                editability === message_edit.editability_types.FULL ||
+                editability === message_edit.editability_types.TOPIC_ONLY;
+            const should_display_quote_and_reply = message.content !== "<p>(deleted)</p>";
+
+            const conversation_time_uri = hash_util.by_conversation_and_time_uri(message);
+
+            const should_display_delete_option = message_edit.get_deletability(message);
+            const args = {
+                message_id,
+                historical: message.historical,
+                stream_id: message.stream_id,
+                topic,
+                use_edit_icon,
+                editability_menu_item,
+                can_mute_topic,
+                can_unmute_topic,
+                should_display_collapse,
+                should_display_uncollapse,
+                should_display_add_reaction_option: message.sent_by_me,
+                should_display_edit_history_option,
+                should_display_hide_option,
+                conversation_time_uri,
+                narrowed: narrow_state.active(),
+                should_display_delete_option,
+                should_display_reminder_option: feature_flags.reminders_in_message_action_menu,
+                should_display_edit_and_view_source,
+                should_display_quote_and_reply,
+            };
+
+            instance.setContent(render_actions_popover_content(args));
+            message_actions_popover_visible = true;
+
+            const $popper = $(instance.popper);
+
+            $popper.one("click", instance.hide);
```

Comment:
How does this work?  It seems like this makes clicking anywhere on the popover's exterior close it, which does't seem like what we want.  Is the intent to just avoid putting `instance.hide()` + `e.stopPropagation()` in each individual click handler?  

If we're doing that, we should comment it clearly.  But I'd also probably be happier with a commit series that did a more exact translation (just moves the code with each element having `instance.hide()` and `e.stopPropagation`) and then a follow-up to try to deduplicate; that'd be much easier to reason about.

Reply:
We want to hide the popover on clicking any button. I had to remove `stopPropagation` since I want the click events to be propagated throughout the popover.